src/params/types.ts

```diff
@@ -183,26 +214,54 @@ export abstract class Param<T extends string | number | boolean | string[]> exte
     throw new Error("Not implemented");
   }
 
-  cmp(cmp: "==" | ">" | ">=" | "<" | "<=", rhs: T) {
+  cmp(cmp: "==" | "!=" | ">" | ">=" | "<" | "<=", rhs: T | Expression<T>) {
     return new CompareExpression<T>(cmp, this, rhs);
   }
 
-  equals(rhs: T) {
+  equals(rhs: T | Expression<T>) {
     return this.cmp("==", rhs);
   }
 
+  notEquals(rhs: T | Expression<T>) {
+    return this.cmp("!=", rhs);
+  }
+
+  greaterThan(rhs: T | Expression<T>) {
+    return this.cmp(">", rhs);
+  }
+
+  greaterThanOrEqualTo(rhs: T | Expression<T>) {
+    return this.cmp(">=", rhs);
+  }
+
+  lessThan(rhs: T | Expression<T>) {
+    return this.cmp("<", rhs);
+  }
+
+  lessThanorEqualTo(rhs: T | Expression<T>) {
+    return this.cmp("<=", rhs);
+  }
+
   toString(): string {
     return `params.${this.name}`;
   }
 
-  toSpec(): ParamSpec<T> {
-    const out: ParamSpec = {
+  toSpec(): WireParamSpec<T> {
+    const { default: paramDefault, ...otherOptions } = this.options;
+
+    const out: WireParamSpec<T> = {
       name: this.name,
-      ...this.options,
+      ...otherOptions,
       type: (this.constructor as typeof Param).type,
     };
 
-    return out as ParamSpec<T>;
+    if (paramDefault instanceof Expression) {
+      out.default = paramDefault.toCEL();
+    } else if (paramDefault) {
```

Comment:
what is `paramDefault = false`? Should this condition be instead `paramDefault !== undefined` or something?

Reply:
yeah, that's what I was meaning to capture